src/App.js

```diff
@@ -0,0 +1,178 @@
+        let urlGenre = `https://api.themoviedb.org/3/genre/movie/list?api_key=${apiKey}&language=en-US`
+        let dataGenre = await fetch(urlGenre)
+        let resultGenre = await dataGenre.json()
+        console.log('genre list:', resultGenre.genres)
+        setGenres(resultGenre.genres)
+        console.log('genre list:', genres)
+    }
+
+    let loadMoreMovie = async (categoryLoadMore) => {
+        page++
+        let url = `https://api.themoviedb.org/3/movie/${categoryLoadMore}?api_key=${apiKey}&language=en-US&page=${page}`
+        let data = await fetch(url)
+        let dataResult = await data.json()
+        setMovies(movies.concat(dataResult.results))
+    }
+
+    let onFilterCategory = async (catergory) => {
+        let url = `https://api.themoviedb.org/3/movie/${catergory}?api_key=${apiKey}&language=en-US&page=1`
+        let data = await fetch(url)
+        let dataResult = await data.json()
+        console.log(dataResult)
+        setMovies(dataResult.results)
+        setcategoryLoadMore(catergory)
+    }
+
+    let onShowVideo = async (movieId) => {
+        let url = `https://api.themoviedb.org/3/movie/${movieId}/videos?api_key=${apiKey}&language=en-US;`
+        let data = await fetch(url)
+        let respones = await data.json()
+        setModal(true)
+        if (respones.results.length > 0) {
+            setMovieVideo(respones.results[0].key)
+        }
+        console.log(respones)
+    }
+
+    let onSearch = (keyword) => {
+        if (keyword === '') {
+            setMovies(movieList)
+        } else {
+            setMovies(movies.filter((movie) => movie.title.toLowerCase().includes(keyword.toLowerCase())))
+        }
+    }
+
+    let onSortByHighestPopularity = () => {
+        let newMovieList = [...movies].sort((a, b) => a.popularity > b.popularity ? -1 : 1)
+        setMovies(newMovieList)
+    }
+
+    let onSortByLowestPopularity = () => {
+        let newMovieList = [...movies].sort((a, b) => a.popularity < b.popularity ? -1 : 1)
+        setMovies(newMovieList)
+    }
+
+    let onFilterRating = (rating) => {
+        console.log(rating)
+        console.log(rating.min, rating.max)
+        setMovies(movieList.filter((movie) => movie.vote_average >= rating.min && movie.vote_average <= rating.max))
+        console.log('rating movie', movies)
+    }
+
+    let handlePageChange = async (pageNumber) => {
+        setPageNumber(pageNumber)
+        let url = `https://api.themoviedb.org/3/movie/${categoryLoadMore}?api_key=${apiKey}&language=en-US&page=${pageNumber}`
+        let data = await fetch(url)
+        let dataResult = await data.json()
+        setMovies(dataResult.results)
+    }
+
+    useEffect(() => {
+        currentPlaying()
+        fetchGenres()
+    }, [])
+
+    // useEffect(() => {
+    //     currentPlaying()
+    // }, [genres]) // put genres inside array mean. hey, if genres change exicute function currentPlaying
+
+
+    if (movies === []) {
+        return (
```

Comment:
later we can work on https://reactjs.org/docs/concurrent-mode-suspense.html :) but for now this is totally fine